libbpf-rs/src/xdp.rs

```diff
@@ -91,4 +92,14 @@ impl<'fd> Xdp<'fd> {
             unsafe { libbpf_sys::bpf_xdp_query_id(ifindex, flags.bits() as i32, &mut prog_id) };
         util::parse_ret(err).map(|()| prog_id)
     }
+
+    /// Replace an existing xdp program (identified by old_prog_fd) with this xdp program
+    pub fn replace(&self, ifindex: i32, old_prog_fd: RawFd, flags: XdpFlags) -> Result<()> {
```

Comment:
We shouldn't be working with a `RawFd` here. Can you please use a primitive with proper lifetime management?